File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-1.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int n = nums.length;
        int window_sum = 0;
        int left = 0;
        int min_len = n + 1;
        for (int right = 0; right < n; right++) {
            window_sum += nums[right];
            while (window_sum >= target) {
                min_len = Math.min(min_len, right - left + 1);
                window_sum -= nums[left];
                left++;
            }
        }
        return min_len <= n ? min_len : 0;
    }
}

File: Data Structures & Algorithms/search-in-rotated-sorted-array-ii/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> bool:
        def check(i: int) -> bool:
            x = nums[i]
            if x > nums[right]:
                return target > nums[right] and x >= target
            return target > nums[right] or x >= target

        left, right = -1, len(nums) - 1 # 开区间（-1,n-1）
        while left + 1 < right:  # 开区间不为空
            mid = (left + right) // 2
            if nums[mid] == nums[right]:
                right -= 1
            elif check(mid):
                right = mid
            else:
                left = mid
        return nums[right] == target
